assets/js/modules/analytics/setup/setup-form.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Analytics Setup form.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../../../components/button';
+import AccountSelect from '../common/account-select';
+import PropertySelect from '../common/property-select';
+import ProfileSelect from '../common/profile-select';
+import { STORE_NAME } from '../datastore';
+import { isValidAccountID, isValidPropertyID } from '../util';
+
+export default function SetupForm() {
+	const accounts = useSelect( ( select ) => select( STORE_NAME ).getAccounts() ) || [];
+	const accountID = useSelect( ( select ) => select( STORE_NAME ).getAccountID() );
+	const propertyID = useSelect( ( select ) => select( STORE_NAME ).getAccountID() );
+	const hasExistingTag = useSelect( ( select ) => select( STORE_NAME ).hasExistingTag() );
+	const isSavingSettings = useSelect( ( select ) => select( STORE_NAME ).isDoingSaveSettings() );
+
+	const isBlockedFromSaving = (
+		isSavingSettings ||
+		! isValidAccountID( accountID ) ||
+		! isValidPropertyID( propertyID )
+	);
```

Comment:
This should also check whether any of the three is empty (e.g. profile could be on "Select a profile...").

Reply:
This part is getting replaced by a new `canSubmitChanges` selector which includes much more logic than is here which not enough.

Reply:
Let's decide for a consistent name though. We talked about an `applyChanges()` action before for applying all changes (maybe that's just `saveSettings()`, but may be more) - in this case, this selector should be `canApplyChanges()`.